tests/integration/aclgen_test.py

```diff
@@ -0,0 +1,181 @@
+      'sample_multitarget.asa',
+      'sample_multitarget.bacl',
+      'sample_multitarget.eacl',
+      'sample_multitarget.ipt',
+      'sample_multitarget.jcl',
+      'sample_multitarget.xacl',
+      'sample_nsxv.nsx',
+      'sample_packetfilter.pf',
+      'sample_speedway.ipt',
+      'sample_srx.srx'
+    ]
+    def makeoutput(f):
+      return 'writing file: {0}'.format(os.path.join(self.output_dir, f))
+
+    actual_output = self.iobuff.getvalue().split("\n")
+    for expected_output in map(makeoutput, expected_files):
+      self.assertTrue(expected_output in actual_output)
+
+    self.assertTrue('writing 15 files to disk...' in actual_output)
+
+
+  def test_generate_single_policy(self):
+    args = [
+      'program',  # Dummy value for gflags, which expects the program name to be the first entry.
+      '--policy_file={0}'.format(os.path.join(self.policies_dir, 'pol', 'sample_cisco_lab.pol')),
+      '--definitions_directory={0}'.format(self.defs_dir),
+      '--output_directory={0}'.format(self.output_dir)
+    ]
+    aclgen.main(args)
+
+    actual_output = self.iobuff.getvalue()
+    expected_outputs = [
+      'rendering one file',
+      os.path.join(self.output_dir, 'sample_cisco_lab.acl')
+    ]
+    for s in expected_outputs:
+      self.assertTrue(s in actual_output)
+
+
+class AclGen_Characterization_Test_Base(unittest.TestCase):
+  """Ensures base functionality works.  Uses data in
+  characterization_data subfolder."""
+
+  def setUp(self):
+    self.iobuff = StringIO()
+    logger = logging.getLogger()
+    logger.level = logging.DEBUG
+    self.s = logging.StreamHandler(self.iobuff)
+    self.s.level = logging.DEBUG
+    logger.addHandler(self.s)
+
+    curr_dir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+    self.test_dir = os.path.join(curr_dir, '..', 'characterization_data')
+    self.output_dir = self.dirpath('filters_actual')
+    if not os.path.exists(self.output_dir):
+      os.makedirs(self.output_dir)
+    self.empty_output_dir(self.output_dir)
+
+  def dirpath(self, *args):
+    return os.path.realpath(os.path.join(self.test_dir, *args))
+
+  def empty_output_dir(self, d):
+    entries = [ os.path.join(d, f) for f in os.listdir(d)]
+    for f in [e for e in entries if os.path.isfile(e)]:
+      os.remove(f)
+    for d in [e for e in entries if os.path.isdir(e)]:
+      shutil.rmtree(d)
+
+
+class AclGen_Arguments_Tests(AclGen_Characterization_Test_Base):
+
+  def test_missing_defs_folder_raises_error(self):
+    def_dir, pol_dir, expected_dir = map(self.dirpath, ('def', 'policies', 'filters_expected'))
+    args = [
+      'program',  # Dummy value for gflags, which expects the program name to be the first entry.
+      '--base_directory={0}'.format(pol_dir),
+      '--definitions_directory=/some_missing_dir/',
+      '--output_directory={0}'.format(self.output_dir)
+    ]
+
+    aclgen.main(args)
+
+    # NOTE that the code still continues work, even if a bad directory
```

Comment:
This is related to something I documented in a closed PR: "Error not raised on naming.NoDefinitionsError in aclgen".
Files are still generated even if there is a bad def’s directory, the error is logged as fatal but swallowed.

```
  except naming.NoDefinitionsError:
    logging.fatal('bad definitions directory: %s', FLAGS.definitions_directory)
```

I didn't think that this was the intended behaviour, and added this test for discussion.  At any rate, this test documents what's currently happening.